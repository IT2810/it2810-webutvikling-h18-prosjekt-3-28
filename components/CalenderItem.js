import React from 'react';
import {
  StyleSheet,
  View,
} from 'react-native';
import {
  Text,
} from 'react-native-elements';

import{
  LinearGradient,
} from 'expo';

class CalenderItem extends React.Component{

    constructor (props) {
      super(props)
      this.state = {
        text: props.text,
        time: props.time,
        location: props.location,
      }
    }
    

    render () {
      return (
        <View style={[styles.item,styles.col]}>
            <View>
            <Text style={[styles.subtitle,styles.titleText3,styles.h4]}>{this.state.time}</Text>
            <Text style={[styles.titleText2,styles.h1]}>{this.state.text}</Text>
            <Text style={[styles.subtitle,styles.titleText3,styles.h4]}>w/ {this.state.location}</Text>
            </View>
        </View>
      )
    }
  }

  
    const styles = StyleSheet.create({
        item: {
            backgroundColor: 'white',
            borderRadius: 10,
            padding: 10,
            marginTop: 17,
            width: 350,
        },
        subtitle:{
            color: '#40434b',
        },
        col:{
          flexDirection: 'row',
        },
        titleText1: {
          fontFamily: 'SF-Pro-Display-Bold',
        },
        titleText2: {
          fontFamily: 'SF-Pro-Display-Regular',
        },
        titleText3: {
          fontFamily: 'SF-Pro-Display-Thin',
        },
        titleText4: {
          fontFamily: 'SF-Pro-Display-Ultralight',
        },
        tab:{
          marginTop:30,
        },
        h1:{
          fontSize:35,
        },
        h2:{
          fontSize:30,
        },
        h3:{
          fontSize:25,
        },
        h4:{
          fontSize:20,
        }

    });

export default CalenderItem;