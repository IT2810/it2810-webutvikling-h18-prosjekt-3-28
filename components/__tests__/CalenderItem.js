import React from 'react';
import CalenderItem from '../CalenderItem';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

describe('App snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
      NavigationTestUtils.resetInternalState();
    });
  
    it('renders the loading screen', async () => {
      const tree = renderer.create(<CalenderItem />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders the root without loading screen', async () => {
      const tree = renderer.create(<CalenderItem skipLoadingScreen />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});

