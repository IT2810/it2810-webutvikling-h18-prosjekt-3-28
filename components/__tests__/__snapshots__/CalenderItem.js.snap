// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders the loading screen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "marginTop": 17,
        "padding": 10,
        "width": 350,
      },
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#40434b",
            },
            Object {
              "fontFamily": "SF-Pro-Display-Thin",
            },
            Object {
              "fontSize": 20,
            },
          ],
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "fontFamily": "SF-Pro-Display-Regular",
            },
            Object {
              "fontSize": 35,
            },
          ],
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#40434b",
            },
            Object {
              "fontFamily": "SF-Pro-Display-Thin",
            },
            Object {
              "fontSize": 20,
            },
          ],
        ]
      }
    >
      w/ 
    </Text>
  </View>
</View>
`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "marginTop": 17,
        "padding": 10,
        "width": 350,
      },
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#40434b",
            },
            Object {
              "fontFamily": "SF-Pro-Display-Thin",
            },
            Object {
              "fontSize": 20,
            },
          ],
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "fontFamily": "SF-Pro-Display-Regular",
            },
            Object {
              "fontSize": 35,
            },
          ],
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#40434b",
            },
            Object {
              "fontFamily": "SF-Pro-Display-Thin",
            },
            Object {
              "fontSize": 20,
            },
          ],
        ]
      }
    >
      w/ 
    </Text>
  </View>
</View>
`;
